controllers/sample_controller_rendered_resources.go

```diff
@@ -0,0 +1,359 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/ghodss/yaml"
+	"github.com/go-logr/logr"
+	"golang.org/x/time/rate"
+	errors2 "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	yamlUtil "k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/kyma-project/template-operator/api/v1alpha1"
+
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/ratelimiter"
+)
+
+const (
+	requeueInterval = time.Second * 3
+	finalizer       = "sample-finalizer"
+	debugLogLevel   = 2
+	fieldOwner      = "sample-field-owner"
+)
+
+var (
+	ConditionTypeInstallation = "Installation"
+	ConditionReasonReady      = "Ready"
+)
+
+// SampleReconciler reconciles a Sample object
+type SampleReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+	*rest.Config
+	// EventRecorder for creating k8s events
+	record.EventRecorder
+}
+
+type RateLimiter struct {
+	Burst           int
+	Frequency       int
+	BaseDelay       time.Duration
+	FailureMaxDelay time.Duration
+}
+
+type ManifestResources struct {
+	Items []*unstructured.Unstructured
+	Blobs [][]byte
```

Comment:
you don't really need the blobs from declarative here

Reply:
I was thinking of removing it, wanted to double check if they can be useful. Removed it.